service/authorization/v2/authorization.go

```diff
@@ -0,0 +1,80 @@
+package authorization
+
+import (
+	"context"
+	"errors"
+
+	"connectrpc.com/connect"
+	authzV2 "github.com/opentdf/platform/protocol/go/authorization/v2"
+	authzV2Connect "github.com/opentdf/platform/protocol/go/authorization/v2/authorizationv2connect"
+	otdf "github.com/opentdf/platform/sdk"
+	"github.com/opentdf/platform/service/logger"
+	"github.com/opentdf/platform/service/pkg/serviceregistry"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Service struct {
+	sdk    *otdf.SDK
+	config *Config
+	logger *logger.Logger
+	trace.Tracer
+}
+
+type Config struct{}
+
+func NewRegistration() *serviceregistry.Service[authzV2Connect.AuthorizationServiceHandler] {
+	as := new(Service)
+
+	return &serviceregistry.Service[authzV2Connect.AuthorizationServiceHandler]{
+		ServiceOptions: serviceregistry.ServiceOptions[authzV2Connect.AuthorizationServiceHandler]{
+			Namespace:      "authorization",
+			Version:        "v2",
+			ServiceDesc:    &authzV2.AuthorizationService_ServiceDesc,
+			ConnectRPCFunc: authzV2Connect.NewAuthorizationServiceHandler,
+			RegisterFunc: func(srp serviceregistry.RegistrationParams) (authzV2Connect.AuthorizationServiceHandler, serviceregistry.HandlerServer) {
+				authZCfg := new(Config)
+
+				logger := srp.Logger
+
+				// default ERS endpoint
+				as.sdk = srp.SDK
+				as.logger = logger
+				// if err := srp.RegisterReadinessCheck("authorization", as.IsReady); err != nil {
+				// 	logger.Error("failed to register authorization readiness check", slog.String("error", err.Error()))
+				// }
```

Comment:
rm

Reply:
Thanks, and I went back and forth on this but I decided I'm going to keep it in for when the readiness check RPC above is uncommented alongside it.